src/Temporalio/Client/TemporalClient.Workflow.cs

```diff
@@ -229,6 +229,15 @@ public override async Task<TResult> QueryWorkflowAsync<TResult>(QueryWorkflowInp
             public async override Task<WorkflowUpdateHandle<TResult>> StartWorkflowUpdateAsync<TResult>(
                 StartWorkflowUpdateInput input)
             {
+                if (input.Options.WaitForStage == WorkflowUpdateStage.None)
+                {
+                    throw new ArgumentException("WaitForStage is required to start workflow update");
+                }
+                else if (input.Options.WaitForStage == WorkflowUpdateStage.Admitted)
```

Comment:
Could just allow the server to reject this call? That way you don't need to update the SDK if the server adds support .

Reply:
Same thing was mentioned at https://github.com/temporalio/sdk-python/pull/521#discussion_r1607125013. I think we need to document what works and what doesn't here anyways, so that'll have to be updated anyways, and hopefully this becomes a default parameter in some durable-update future.